model/admin/reports.php

```diff
@@ -25,22 +25,33 @@ public function zap_report()
     {
         global $lang_admin_reports;
 
-        
-
         $zap_id = intval(key($this->request->post('zap_id')));
 
-        $result = $this->db->query('SELECT zapped FROM '.$this->db->prefix.'reports WHERE id='.$zap_id) or error('Unable to fetch report info', __FILE__, __LINE__, $this->db->error());
-        $zapped = $this->db->result($result);
-
-        if ($zapped == '') {
-            $this->db->query('UPDATE '.$this->db->prefix.'reports SET zapped='.time().', zapped_by='.$this->user->id.' WHERE id='.$zap_id) or error('Unable to zap report', __FILE__, __LINE__, $this->db->error());
+        $result = \ORM::for_table($this->feather->prefix.'reports')
+            ->select('zapped')
+            ->where('id', $zap_id)
+            ->find_one_col('zapped');
+
+        if (!$result) {
+            \ORM::for_table($this->feather->prefix.'reports')
+                ->where('id', $zap_id)
+                ->find_one()
+                ->set(array('zapped' => time(),
```

Comment:
Can you create a new $var = array(...) ? It will be easier to alter it with hooks this way :)